### PR TITLE
Allow to pass full config to k0s container as env

### PR DESCRIPTION
**Pull request by jakolehm**
**What this PR Includes**

An easier way to pass k0s config.yaml to k0s container.

### CODE REVIEW

**Review by jasmingacic — APPROVED** [review]

**Review by mviitane — COMMENTED** [review]
Chapter "Running workers" still includes 
docker.pkg.github.com/k0sproject/k0s/k0s:<version>

It should be changed and aligned with the rest of the document.

File: docs/k0s-in-docker.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ We publish a k0s container image with every release. By default, we run both con
 
 You can run your own k0s-in-docker easily with:
 ```sh
-docker run -d --name k0s --hostname k0s --privileged -v /var/lib/k0s -p 6443:6443 docker.pkg.github.com/k0sproject/k0s/k0s:<version>
+docker run -d --name k0s --hostname k0s --privileged -v /var/lib/k0s -p 6443:6443 docker.io/k0sproject/k0s:latest
```
Comment (ncopa) [inline docs/k0s-in-docker.md:7]: What is the reason for this change?

Comment (jnummelin) [reply docs/k0s-in-docker.md:7]: Docker Hub images pull does not need and auth whereas Github registry does always need auth. Easier for the users

**Review by jnummelin — APPROVED** [review]
